tests/reference_images.rs

```diff
@@ -95,49 +97,135 @@ fn check_references() {
-            // This might happen because the testsuite contains unsupported images
-            // or because a specific decoder included via a feature.
-            Err(image::ImageError::UnsupportedError(_)) => return,
-            Err(err) => panic!(format!("decoding of {:?} failed with: {}", path, err)),
-        };
-        let mut test_crc = Crc32::new();
-        test_crc.update(&*test_img);
-        if *ref_img != *test_img || test_crc.finalize() != ref_crc {
-            panic!(
-                "Reference rendering does not match for image at {:?}.",
-                img_path
-            )
+        img_path.push(orig_filename);
+
+        // Load the test image
+        let test_img;
+
+        if let Some(anim_frame) = anim_frame {
+            #[cfg(feature = "gif_codec")]
+            {
+                // Interpret the input file as an animation file and extract a
+                // single frame
+                use image::AnimationDecoder;
+                let stream = io::BufReader::new(fs::File::open(&img_path).unwrap());
+                let decoder = match image::gif::Decoder::new(stream) {
+                    Ok(decoder) => decoder,
+                    Err(image::ImageError::UnsupportedError(_)) => return,
+                    Err(err) => panic!(format!("decoding of {:?} failed with: {}", img_path, err)),
+                };
+
+                let mut frames = match decoder.into_frames().collect_frames() {
+                    Ok(frames) => frames,
+                    Err(image::ImageError::UnsupportedError(_)) => return,
+                    Err(err) => panic!(format!(
+                        "collecting frames of {:?} failed with: {}",
+                        img_path, err
+                    )),
+                };
+
+                let frame = frames.drain(anim_frame..).nth(0).unwrap();
+
+                // Convert the frame to a`RgbaImage`
+                test_img = frame.into_buffer();
+            }
+
+            #[cfg(not(feature = "gif_codec"))]
+            {
+                println!("Skipping - GIF codec is not enabled");
+                return;
+            }
+        } else {
+            // Read the input file as a single image
+            match image::open(&img_path) {
+                Ok(img) => test_img = img.to_rgba(),
+                // Do not fail on unsupported error
+                // This might happen because the testsuite contains unsupported images
+                // or because a specific decoder included via a feature.
+                Err(image::ImageError::UnsupportedError(_)) => return,
+                Err(err) => panic!(format!("decoding of {:?} failed with: {}", img_path, err)),
+            };
+        }
+
+        let ref_img = coalesce_transparent_pixels(ref_img);
+        let test_img = coalesce_transparent_pixels(test_img);
+
+        //ref_img.save("/tmp/test-ref.png").unwrap();
+        //test_img.save("/tmp/test-in.png").unwrap();
+
+        if *ref_img != *test_img {
```

Comment:
These three lines above relaxes the tests on all images: No longer compares exact checksums and doesn't compare visually transparent pixels. If for some reason, the strict tests can not be applied to animated `gif` then it would make sense to have a seperate set of assertions for them instead of modifying the assertions for all other images.